pym/bob/generators/QtCreatorGenerator.py

```diff
@@ -239,18 +251,28 @@ def qtProjectGenerator(package, argv, extra):
 
     # now go through all checkout dirs to find all header / include files
     for name,path in OrderedDict(sorted(dirs, key=lambda t: t[1])).items():
-        newPath = os.path.join(symlinkDir, name)
-        if not args.update: os.symlink(os.path.join(os.getcwd(), path), newPath)
+        if isWindows():
+            name = name.replace('::', '__')
+            newPath = os.path.join(cwd(), path)
+        else:
+            newPath = os.path.join(symlinkDir, name)
+            if not args.update: os.symlink(os.path.join(cwd(), path), newPath)
         for root, directories, filenames in os.walk(newPath):
             hasInclude = False
             if (((os.path.sep + '.git' + os.path.sep) in root) or
                 ((os.path.sep + '.svn' + os.path.sep) in root)):
                 continue
             for filename in filenames:
                 if source.match(filename) or cmake.match(filename) or filename == 'CMakeLists.txt':
-                    sList.append(os.path.join(os.getcwd(), os.path.join(root,filename)))
+                    if isWindows():
+                        sList.append(os.path.join(root,filename))
+                    else:
+                        sList.append(os.path.join(cwd(), os.path.join(root,filename)))
```

Comment:
Oh, I guess this will be slow on Windows with many files. What about calling `cwd()` once at the start of `qtProjectGenerator()` and use a variable to store the value?

Reply:
I am not a pyhton geek, would that be okay @jkloetzke?

`
# helper to get linux or windows (MSYS2 support) cwd
+pwd = None
 def cwd():
-    return os.getcwd() if not isWindows() else os.popen('pwd -W').read().strip()
+    global pwd
+    if pwd == None and isWindows():
+        pwd = os.popen('pwd -W').read().strip()
+    return os.getcwd() if not isWindows() else pwd
`